core/state_processor.go

```diff
@@ -166,7 +184,6 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 			} else {
 				// Update the pointers in the forking db
 				fdb.LastAccHash = accIt.Hash()
```

Comment:
Uhm, isn't this line suffering from a side-effect caused by L182?
While trying to detect if we finish with the conversion, we called `actIt.Next()`. If there's still work to do it will return `true` _and move the iterator one place forward_. And when executing L186, `accIt.Hash()` will save as the "last processed" the "next" one instead of the last processed one. That doesn't sound right if the last processed one still had storage-slots to migrate.